python/cudf/dataframe/index.py

```diff
@@ -117,16 +118,151 @@ def _concat(cls, objs):
         data = Column._concat([o.as_column() for o in objs])
         return as_index(data)
 
-    def __eq__(self, other):
-        if not isinstance(other, Index):
+    def _binaryop(self, other, fn):
+        """
+        Internal util to call a binary operator *fn* on operands *self*
+        and *other*.  Return the output Series.  The output dtype is
+        determined by the input operands.
+        """
+        from cudf import DataFrame, Series
+        idx_series = Series(self)
```

Comment:
I feel like you should just call `get_attr(idx_series, fn)(other)` after this line for `_binaryop`, `_rbinaryop`, and `_unaryop`. `Series` already has all these functions.

Reply:
Going to refactor to just have one helper function to handle converting to series, calling the binary op, then converting back to index.